miden/tests/integration/operations/crypto_ops.rs

```diff
@@ -10,6 +10,26 @@ use crate::helpers::crypto::{init_merkle_leaf, init_merkle_leaves};
 // TESTS
 // ================================================================================================
 
+#[test]
+fn hash() {
+    let asm_op = "hash";
+
+    // --- test hashing 4 random values -----------------------------------------------------------
+    let mut values = Vec::with_capacity(8);
+
+    // push 4 random elements to be hashed
+    let random_values = rand_vector::<u64>(4);
+    values.extend_from_slice(&random_values);
+    // pad 4 remaining rate elements
+    values.extend_from_slice(&[0, 0, 0, 0]);
+    let expected = build_expected_hash(&values);
```

Comment:
Since we are hashing just 4 elements, we shouldn't need to pad the input with zeros. The could could look like this:
```Rust
// push 4 random elements to be hashed
let random_values = rand_vector::<u64>(4);
let expected = build_expected_hash(&random_values);
```